File: app/src/main/java/com/example/chgadapter_kotlin_demo/weibo/activity/ShowBigImageViewActivity.java
//package com.example.chgadapter_kotlin_demo.weibo.activity;
//
//import android.os.Bundle;
//
//import androidx.annotation.Nullable;
//import androidx.appcompat.app.AppCompatActivity;
//import androidx.recyclerview.widget.LinearLayoutManager;
//import androidx.recyclerview.widget.PagerSnapHelper;
//import androidx.recyclerview.widget.RecyclerView;
//
//import com.chg.adapter.EventTransmissionListener;
//import com.chg.adapter.Model;
//import com.example.chgadapter_kotlin_demo.R;
//
//import java.util.HashMap;
//import java.util.List;
//
//public class ShowBigImageViewActivity extends AppCompatActivity {
//    private RecyclerView recyclerView;
//    @Override
//    protected void onCreate(@Nullable Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        setContentView(R.layout.show_big_image_view);
//
//        recyclerView = findViewById(R.id.recyclerView);
//
//        LinearLayoutManager manager = new LinearLayoutManager(ShowBigImageViewActivity.this);
//        manager.setOrientation(LinearLayoutManager.HORIZONTAL);
//        recyclerView.setLayoutManager(manager);
//
//        PagerSnapHelper snapHelper = new PagerSnapHelper();
//        snapHelper.attachToRecyclerView(recyclerView);
//
//
//        HashMap map = (HashMap) getIntent().getExtras().get("sources");
//        recyclerView.setModels((List<Model>) map.get("sources"));
//        recyclerView.scrollToPosition((Integer) map.get("position"));
//        recyclerView.setEventTransmissionListener(new EventTransmissionListener() {
//            @org.jetbrains.annotations.Nullable
//            @Override
//            public Object onEventTransmissionListener(@org.jetbrains.annotations.Nullable Object target, @org.jetbrains.annotations.Nullable Object params, int eventId, @org.jetbrains.annotations.Nullable CallBack callBack) {
//                return null;
//            }
//
//        });
//
//
//    }
//}

File: app/src/main/java/com/example/chgadapter_kotlin_demo/weibo/activity/ShowBigImageViewActivity.kt
package com.example.chgadapter_kotlin_demo.weibo.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.PagerSnapHelper
import androidx.recyclerview.widget.RecyclerView
import com.chg.adapter.EventTransmissionListener
import com.chg.adapter.EventTransmissionListener.CallBack
import com.chg.adapter.Model
import com.chg.adapter.extension.eventTransmissionListener
import com.chg.adapter.extension.getContext
import com.chg.adapter.extension.models
import com.example.chgadapter_kotlin_demo.R
import java.util.*

class ShowBigImageViewActivity : AppCompatActivity(),EventTransmissionListener {

    private lateinit var recyclerView: RecyclerView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.show_big_image_view)
        recyclerView = findViewById(R.id.recyclerView)
        val manager = LinearLayoutManager(getContext())
        manager.orientation = LinearLayoutManager.HORIZONTAL
        recyclerView.setLayoutManager(manager)
        val snapHelper = PagerSnapHelper()
        snapHelper.attachToRecyclerView(recyclerView)
        val map:MutableMap<Any,Any> = intent.extras!!["sources"] as MutableMap<Any, Any>
        recyclerView.models = map["sources"] as List<Model>?
        recyclerView.scrollToPosition((map!!["position"] as Int?)!!)
        recyclerView.eventTransmissionListener = this

    }

    override fun onEventTransmissionListener(target: Any?, params: Any?, eventId: Int, callBack: CallBack?): Any? {
        finish()
        return null
    }
}